File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppStorage.java
package com.iterable.iterableapi;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.List;

interface IterableInAppStorage {
    @NonNull
    List<IterableInAppMessage> getMessages();

    @Nullable
    IterableInAppMessage getMessage(String messageId);

    void addMessage(@NonNull IterableInAppMessage message);

    void removeMessage(@NonNull IterableInAppMessage message);

    void saveHTML(@NonNull String messageID, @NonNull String contentHTML);

    @Nullable
    String getHTML(@NonNull String messageID);

    void removeHTML(@NonNull String messageID);
}

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppStorage.kt
package com.iterable.iterableapi

import androidx.annotation.NonNull
import androidx.annotation.Nullable

internal interface IterableInAppStorage {
    @NonNull
    fun getMessages(): List<IterableInAppMessage>

    @Nullable
    fun getMessage(messageId: String): IterableInAppMessage?

    fun addMessage(@NonNull message: IterableInAppMessage)

    fun removeMessage(@NonNull message: IterableInAppMessage)

    fun saveHTML(@NonNull messageID: String, @NonNull contentHTML: String)

    @Nullable
    fun getHTML(@NonNull messageID: String): String?

    fun removeHTML(@NonNull messageID: String)
}
